internal/server/server.go

```diff
@@ -31,53 +29,41 @@ func NewNeo4jMCPServer(version string, cfg *config.Config) (*Neo4jMCPServer, err
 			"by inferring the schema with tools like get-schema and executing arbitrary Cypher queries with read-cypher."),
 	)
 
-	// Initialize Neo4j driver once
-	driver, err := neo4j.NewDriverWithContext(cfg.URI, neo4j.BasicAuth(cfg.Username, cfg.Password, ""))
-	if err != nil {
-		wrappedErr := fmt.Errorf("failed to create Neo4j driver: %w", err)
-		log.Printf("Error in NewNeo4jMCPServer: %v", wrappedErr)
-		return nil, wrappedErr
-	}
-
 	return &Neo4jMCPServer{
 		mcpServer: mcpServer,
 		config:    cfg,
-		driver:    &driver,
+		dbService: dbService,
 		version:   version,
-	}, nil
+	}
 }
 
 // RegisterTools registers all available MCP tools
 func (s *Neo4jMCPServer) RegisterTools() error {
-	// Create the database service
-	dbService := database.NewNeo4jService(s.driver)
-
 	deps := &tools.ToolDependencies{
 		Config:    s.config,
-		DBService: dbService,
+		DBService: s.dbService,
 	}
 	tools.RegisterAllTools(s.mcpServer, deps)
 	return nil
 }
 
 // Start initializes and starts the MCP server using stdio transport
-func (s *Neo4jMCPServer) Start(ctx context.Context) error {
+func (s *Neo4jMCPServer) Start() error {
 	log.Println("Starting Neo4j MCP Server...")
 
-	// Test the database connection
-	if err := (*s.driver).VerifyConnectivity(ctx); err != nil {
-		return fmt.Errorf("failed to verify database connectivity: %w", err)
-	}
-
 	// Register tools
 	if err := s.RegisterTools(); err != nil {
 		return fmt.Errorf("failed to register tools: %w", err)
 	}
 	log.Println("Started Neo4j MCP Server. Now listening for input...")
+	// Note: ServeStdio handles its own signal management for graceful shutdown
 	return server.ServeStdio(s.mcpServer)
 }
 
-// Stop gracefully stops the server and closes the driver
-func (s *Neo4jMCPServer) Stop(ctx context.Context) error {
-	return (*s.driver).Close(ctx)
+// Stop gracefully stops the server
+func (s *Neo4jMCPServer) Stop() error {
```

Comment:
the context in Start()/Stop() and handler contexts are different:

Start context: Created in main.go, used only for startup connectivity verification
Handler contexts: Created by MCP library per-request, used for actual database operations

since Start() and Stop() functions of the server only use context for one database connectivity check (which we already do in main.go), I just removed it.